pandas/tests/indexes/common.py

```diff
@@ -205,6 +205,39 @@ def test_hash_error(self):
                                        type(ind).__name__):
                 hash(ind)
 
+    def test_copy(self):
+        # GH12309
+        for name, index in compat.iteritems(self.indices):
+            if isinstance(index, MultiIndex):
+                continue
+
```

Comment:
can you add the equivalent of these 2 tests for the ones you are skipping (e.g. MultiIndex) and PI, RI, MI